apps/indexer/src/api/services/delegation-percentage/delegation-percentage.service.ts

```diff
@@ -0,0 +1,339 @@
+import { MetricTypesEnum } from "@/lib/constants";
+import {
+  DelegationPercentageRepository,
+  DaoMetricRow,
+} from "@/api/repositories/delegation-percentage.repository";
+import type {
+  DelegationPercentageItem,
+  PageInfo,
+  DelegationPercentageResponse,
+} from "@/api/mappers/delegation-percentage";
+
+export interface DelegationPercentageFilters {
+  after?: string;
+  before?: string;
+  startDate?: string;
+  endDate?: string;
+  orderDirection?: "asc" | "desc";
+  limit?: number;
+}
+
+interface DateData {
+  delegated?: bigint;
+  total?: bigint;
+  daoId?: string;
+  tokenId?: string;
+}
+
+export class DelegationPercentageService {
+  constructor(private readonly repository: DelegationPercentageRepository) {}
+
+  /**
+   * Main method to get delegation percentage data with forward-fill and pagination
+   */
+  async getDelegationPercentage(
+    filters: DelegationPercentageFilters,
+  ): Promise<DelegationPercentageResponse> {
+    const {
+      after,
+      before,
+      startDate,
+      endDate,
+      orderDirection = "asc",
+      limit = 100,
+    } = filters;
+
+    // 1. Get initial values for proper forward-fill
+    const referenceDate = startDate || after;
```

Comment:
double check this pls

```suggestion
    const referenceDate =  after || startDate;
```

Reply:
marked "resolved" with no address;